source/bayes_simulation.Rmd

```diff
@@ -45,112 +45,283 @@ correctly identifies 80 percent as "OK" but says that 20 percent are
 
--OR-
+We saw that there are two paths in the tree that lead to a mechanic
+approving a car, and we can calculate the fraction of cars that end up so:
 
-1.  Choose a number randomly between "1" and "100."
+1. Of all cars, 30% are faulty.  Of these, 10% are approved by a mechanic.  That is,
+   30% * 10% = 3% of all cars.
+2. Of all cars, 70% work.  Of these, 80% are approved by a mechanic.  That is,
+   70% * 80% = 56% of all cars.
 
-2.  If "28-30" or "31-45:, record; otherwise draw another number, and
-    repeat until a number is recorded.
+The percentage of faulty cars, out of approved cars, is therefore:
 
-3.  Repeat step 2 and count the proportion "28-30" among the total
-    "28-45."
+$$ 3% / (56% + 3%) = 5.08%. $$
 
-The key modeling step is excluding a trial from consideration (without
-making any record of it) if it produces an irrelevant observation, and
-continuing to do so until the process produces an observation of the
-sort about which you are presently inquiring.
+Notation-wise, it is a bit easier to calculate these sums as fractions instead of as percentages:
 
-Using RESAMPLING STATS, an answer may be produced as follows:
+1. Faulty cars approved by a mechanic: 0.3 * 0.1 = 0.03
+2. Working cars approved by a mechanic: 0.7 * 0.8 = 0.56
 
-"01 - 27" = actually faulty, said to be "faulty" "28 - 30 = faulty, said
-to be "OK"
+Fraction of faulty cars out of approved cars: 0.03 / (0.03 + 0.56) = 0.0508
 
-"31 - 86" = OK, "OK"
+**Calculation using simulation:**
 
-"87 - 100" = OK, "faulty"
+1. We note that we are only interested in outcomes where the mechanic says the cars are OK.
 
-`           REPEAT 1000         `
+2. We start with N cars.  For each car, generate a label of either
+   "faulty" or "working" with 0.3 and 0.7 probabilities respectively.
 
-Do 1000 repetitions
+3. For each faulty car, we generate one of two labels, "mechanic
+   approved" or "mechanic failed", with probabilities of 0.1 or 0.9
+   respectively.
 
-`           GENERATE 1 1,100 a         `
+4. For each working car, generate one of two labels, "mechanic
+   approved" or "mechanic failed", with probabilities 0.7 and 0.3
+   respectively.
 
-Generate a number between "1" and "100"
+5. Out of all cars "mechanic approved", count how many are "faulty".
+   The ratio between these numbers is our answer.
 
-`           IF a between 28 86         `
+Here is the code:
 
-If it's between "28" and "86" (those that say "good")
+```{python}
+import numpy as np
 
-`           SCORE a z         `
+N = 100000  # number of cars
 
-Score this number
+# Generate an array with as many entries as there are cars, each
+# being either 'working' or 'faulty'
+cars = np.random.choice(['working', 'faulty'], p=[0.7, 0.3], size=N)
 
-`           END         `
+# Count how many cars are working and how many are faulty
+N_working = np.count_nonzero(cars == 'working')
+N_faulty = N - N_working
 
-End the IF condition
+# Create a new array in which to store mechanic approved (True)
```

Comment:
Or use strings?  As in `mechanic_says` and 'approved', 'not approved'.   Just to avoid the question - why True False Here, and not for the `cars` variable?